src/ingest-pipeline/misc/tools/split_and_create.py

```diff
@@ -338,31 +369,36 @@ def reorganize(source_uuid, **kwargs) -> None:
     frozen_df_fname = kwargs["frozen_df_fname"]
     verbose = kwargs.get("verbose", False)
     dag_config = {}
+    child_uuid_list = []
+    is_multiassay = False
 
     entity_factory = EntityFactory(auth_tok, instance=instance)
 
     print(f"Decomposing {source_uuid}")
     source_entity = entity_factory.get(source_uuid)
+    full_entity = SoftAssayClient(
+        list(source_entity.full_path.glob("*metadata.tsv")), instance, auth_tok
+    )
     if mode in ["all", "stop"]:
-        if hasattr(source_entity, "data_types"):
+        if hasattr(source_entity, "dataset_type"):
+            # ToDo: review this change to dataset_type
             assert isinstance(source_entity.data_types, str)
             source_data_types = source_entity.data_types
         else:
             source_data_types = None
-        source_metadata_files = list(source_entity.full_path.glob("*metadata.tsv"))
-        for src_idx, smf in enumerate(source_metadata_files):
+        for src_idx, smf in enumerate(full_entity.primary_assay.get("metadata-file")):
```

Comment:
Do we anticipate that there will be multiple metadata files in full_entity.primary_assay["metadata-file"]? SoftAssayClient doesn't seem to allow for that. Trying to figure out if this enumeration is still necessary here and on 415 (but admittedly this code is a little tangled for me).

Reply:
We probably won't, this was a straight forward replacement of the file reading with the new multiassay primary. It doesn't break anything leaving it as is (it will just have 0 as index). I agree, that part of the code is quite tangled, maybe we can ask @jswelling what the intention was of having that index in place.

Reply:
Also agreed that this is probably just an aesthetic / clarity concern rather than a functional one.

Reply:
It used to be legit to have multiple (non-multiome) metadata.tsv files in the same upload directory, so that enumeration was meant to cope with them all in order.  Each would control distinct data subdirectories so there was basically no conflict.  I think we've deprecated the notion of multiple metadata.tsv files in cases which are not multiome, though.

Reply:
I don't think that the validation routines will report an error if there are multiple metadata.tsv files for non-multiome uploads. @gesinaphillips - is that the case?

Reply:
I take it back, I actually don't think that this is possible for non-multiome uploads as none of our metadata extraction classes allow for more than a single metadata.tsv file to be present in the root of an upload/dataset.

Reply:
A mixed upload will pass validation in most cases. However, this is currently not the case for a multi-assay upload, which will break if you include *-metadata.tsv files for any other dataset_types EXCEPT its children, which can refer to datasets that are not part of the parent upload (sorry, confusing).

So while this Visium example in my comment below is valid:

```
visium-metadata.tsv
histology-metadata.tsv
rnaseq-metadata.tsv
dataset-referred-to-by-all-metadata-tsvs/
dataset-only-referred-to-by-histology/
```

...something like this would not be:

```
visium-metadata.tsv
histology-metadata.tsv
rnaseq-metadata.tsv
codex-metadata.tsv
dataset-referred-to-by-all-metadata-tsvs-except-codex/
dataset-only-referred-to-by-histology/
dataset-only-referred-to-by-codex/
```

In the above case, you would get the error `Invalid child assay type(s) for parent type Visium (no probes): ["CODEX"].`.

I think that if you can submit multiple dataset types in a given upload then the multi-assay case above should be fixed, but that's how it works right now.

Reply:
As per conversation on 01/05 - we will not support mixed uploads (IE - 10x and single RNAseq datasets). Tasks added for detecting these situations during validation.